tests/testsuite/build.rs

```diff
@@ -2147,6 +2147,40 @@ fn crate_library_path_env_var() {
     setenv_for_removing_empty_component(p.cargo("run")).run();
 }
 
+// See https://github.com/rust-lang/cargo/issues/14194
+#[cargo_test]
+fn issue_14194_deduplicate_library_path_env_var() {
+    let p = project()
+        .file(
+            "src/main.rs",
+            &format!(
+                r#"
+                    use std::process::Command;
+                    fn main() {{
+                        if std::env::var("SUB").is_ok() {{
+                            let paths = match std::env::var_os("{}") {{
+                                Some(var) => std::env::split_paths(&var).collect(),
+                                None => Vec::new(),
+                            }};
+                            let first = paths.first().unwrap();
+                            assert!(paths.iter().skip(1).any(|p| p == first ));
```

Comment:
Should we test for the sympton rather than the specific detail?  In this case, its an extra rebuild